src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx

```diff
@@ -20,12 +20,17 @@ import { useErrorModal } from 'hooks/useErrorMessageAndModal'
 import { tryParseAmount } from 'state/swap/hooks'
 import { calculateInvestmentAmounts, calculatePercentage } from 'state/claim/hooks/utils'
 import { AMOUNT_PRECISION, PERCENTAGE_PRECISION } from 'constants/index'
+import { useGasPrices } from 'state/gas/hooks'
+import { _estimateTxCost } from 'components/swap/EthWethWrap/helpers'
+import { useWalletInfo } from 'hooks/useWalletInfo'
 
 const ErrorMsgs = {
   InsufficientBalance: (symbol = '') => `Insufficient ${symbol} balance to cover investment amount`,
   OverMaxInvestment: `Your investment amount can't be above the maximum investment allowed`,
   InvestmentIsZero: `Your investment amount can't be zero`,
   NotApproved: (symbol = '') => `Please approve ${symbol} token`,
+  InsufficientNativeBalance: (symbol = '', action = "won't") =>
+    `You ${action} have enough ${symbol} to pay the network transaction fee`,
```

Comment:
why not just an enum and 2 different errors for EOA vs SC wallet instead of this fn. We could scrap symbol also since it's obvious that on that error display what the native token is

Reply:
> We could scrap symbol also since it's obvious that on that error display what the native token is

I'd keep it for clarity.